### PR TITLE
#2537-pii-wrapping-at-endpoint

### PR DESCRIPTION
**Pull request by mchlwellman**
# Description

issue #2537 

This pull request implements PII (Personally Identifiable Information) wrapping functionality at the API endpoint entry point, allowing recipient identifiers to be automatically wrapped in appropriate PII classes when the feature flag is enabled.

Key changes:
- Added a new wrap_recipient_identifier_in_pii function that maps identifier types to their corresponding PII classes
- Integrated PII wrapping into the notification endpoint with feature flag control

## How Has This Been Tested?

- [Deployed to dev](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/16651300038)
- Sent notifications
```http
POST /vanotify/v2/notifications/email HTTP/1.1
Content-Type: application/json
Accept: */*
Host: dev-xxx
 {
  "template_id": "ace44b6d-312d-4ce0-afa6-7d77c7be2811",
  "email_address": "xxx@gmail.com"
}
 
HTTP/1.1 201 CREATED
Date: Thu, 31 Jul 2025 14:21:18 GMT
Content-Type: application/json
Content-Length: 639 
{
"billing_code": null,
"content": {
"body": "Hello,\n\nTesting a custom reply-to address.\n\nWith love,\nQA",
"subject": "Tell us your thoughts"
},
"id": "3710efb7-f4bb-4dd9-a7df-15ab81afc0e4",
"reference": null,
"scheduled_for": null,
"template": {
"id": "ace44b6d-312d-4ce0-afa6-7d77c7be2811",
"uri": "https://dev-api.va.gov/services/c6340735-40fb-4ed8-ada1-1cbc34cdfaf0/templates/ace44b6d-312d-4ce0-afa6-7d77c7be2811",
"version": 1
},
"uri": "https://xxx/v2/notifications/3710efb7-f4bb-4dd9-a7df-15ab81afc0e4"
}
```

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

File: tests/app/v2/notifications/test_pii_wrapping.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        """Test that edge cases are handled gracefully."""
+        with notify_api.app_context():
+            original_form = form.copy()
+            result = wrap_recipient_identifier_in_pii(form)
+
+            # Form should be unchanged for all edge cases
+            assert result == original_form
+
+    def test_wrap_recipient_identifier_unknown_id_type(self, notify_api):
+        """Test that unknown id_type is handled gracefully with warning log."""
```
Comment (ChrisJohnson-CDJ) [inline tests/app/v2/notifications/test_pii_wrapping.py:54]: Warning log mentioned in test comment but not actually checked

File: app/v2/notifications/post_notifications.py [hunk 2]
```diff
@@ -47,6 +50,55 @@
...
+    Wrap the recipient identifier id_value in the appropriate PII class.
+
+    Args:
+        form: The validated form data containing recipient_identifier
+
+    Returns:
+        The form with id_value wrapped in appropriate PII class
+    """
+    recipient_identifier = form.get('recipient_identifier')
+    if not recipient_identifier:
```
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/v2/notifications/post_notifications.py]: Actually, it's a dictionary value, but an empty dictionary also would not pass validation.

File: app/v2/notifications/post_notifications.py [hunk 3]
```diff
@@ -47,6 +50,70 @@
...
+    """
+    recipient_identifier: dict | None = form.get('recipient_identifier')
+    if recipient_identifier is None:
+        return form
+
+    id_type: str = recipient_identifier.get('id_type')
+    id_value: str = recipient_identifier.get('id_value')
+
+    if id_type is None or id_value is None:
+        return form
```
Comment (kalbfled) [inline app/v2/notifications/post_notifications.py:86]:
```suggestion
    # These values must be present to pass validation.
    id_type: str = recipient_identifier['id_type']
    id_value: str = recipient_identifier['id_value']
```

If you don't want to trust the validation, you could catch KeyError and use that to log that a programming error exists.  I think the above is adequate.

Comment (mchlwellman) [reply app/v2/notifications/post_notifications.py]: So we are restricting this method to only be used by the v2/post_notifications route?

**Review by kalbfled — APPROVED** [review]

**Review by ChrisJohnson-CDJ — APPROVED** [review]

**Review by ChrisJohnson-CDJ — APPROVED** [review]
